sympy/polys/rings.py

```diff
@@ -2189,8 +2192,17 @@ def _gcd(f, g):
             return f._gcd_QQ(g)
         elif ring.domain.is_ZZ:
             return f._gcd_ZZ(g)
-        else: # TODO: don't use dense representation (port PRS algorithms)
-            return ring.dmp_inner_gcd(f, g)
+        else:
+            if not ring.domain.is_Field:
+                h = _gcd_prs(f, g)
+                cff = f.quo(h)
+                cfg = g.quo(h)
+                return h, cff, cfg
+            else:
+                h = _gcd_prs(f, g).monic()
+                cff = f.quo(h)
+                cfg = g.quo(h)
+                return h, cff, cfg
```

Comment:
Here you are changing the call to `dmp_inner_gcd` and replacing it with a call to the new code. Any differences or test failures anywhere else result from this change. I suggest adding some code here like:
```python
if new_result != old_result:
    raise RuntimeError(f"gcd is different for\n\nf:\n{f}:\n\ng:\n{g} \n\n Old result = ...")
```
Then any case where the new code does not match the old code can be investigated. I don't think that the change from dense to sparse should change the actual computed gcd so I think that any difference is a bug in the new code.